File: android/src/main/java/pub/doric/library/DoricMobxLibrary.java
package pub.doric.library;

import pub.doric.DoricComponent;
import pub.doric.DoricLibrary;
import pub.doric.DoricRegistry;
import pub.doric.utils.DoricUtils;

@DoricComponent
public class DoricMobxLibrary extends DoricLibrary {
    @Override
    public void load(DoricRegistry registry) {
        {
            String content = DoricUtils.readAssetBinFile("bundle_mobx-doric.js");
            registry.registerJSBundle("mobx-doric", content);
        }
        {
            String content = DoricUtils.readAssetBinFile("bundle_mobx.js");
            registry.registerJSBundle("mobx", content);
        }
    }
}

File: iOS/Classes/DoricMobxLibrary.m
#import "DoricMobxLibrary.h"

@implementation DoricMobxLibrary
- (void)load:(DoricRegistry *)registry {
    NSString *path = [[NSBundle mainBundle] bundlePath];
    {
        NSString *fullPath = [path stringByAppendingPathComponent:@"bundle_mobx-doric.js"];
        NSString *jsContent = [NSString stringWithContentsOfFile:fullPath encoding:NSUTF8StringEncoding error:nil];
        [registry registerJSBundle:jsContent withName:@"mobx-doric"];
    }
    {
        NSString *fullPath = [path stringByAppendingPathComponent:@"bundle_mobx.js"];
        NSString *jsContent = [NSString stringWithContentsOfFile:fullPath encoding:NSUTF8StringEncoding error:nil];
        [registry registerJSBundle:jsContent withName:@"mobx"];
    }
}
@end
